lib/hammer_cli_foreman/api/session_authenticator_wrapper.rb

```diff
@@ -57,12 +53,11 @@ def authenticate(request, args)
       end
 
       def error(ex)
-        load_session
-        if ex.is_a?(RestClient::Unauthorized) && !@session_id.nil?
+        if ex.is_a?(RestClient::Unauthorized) && !session.valid?
```

Comment:
`&& session.valid?` the session needs to be valid before we destroy it. When the session is not valid in time we get Unauthorized error we probably provided wrong credentials or there is some other reason for 401.